### ISSUE TITLE
Parsing issues when body contains a single quote

### ISSUE THREAD

**Issue by avin45h** [highlightjs/highlightjs-curl#12]
Hi
Curl parsing is failing when request body contains a single quote

```curl
curl -X POST -H 'Content-Type: application/json' -H 'key: <redacted>' --data '{
  "instance": "{\n  \"conversation_id\": \"sambaverse-conversation-id\",\n  \"messages\": [\n    {\n      \"message_id\": 0,\n      \"role\": \"user\",\n      \"content\": \"write a letter\"\n    },\n    {\n      \"message_id\": 1,\n      \"role\": \"assistant\",\n      \"content\": \"to a friend about a problem'\''? ANSWER: How to Make a Letter to a Friend About a Problem\"\n    },\n    {\n      \"message_id\": 2,\n      \"role\": \"user\",\n      \"content\": \"write a letterto a friend about a problem'\''? ANSWER: How to Make a Letter to a Friend About a Problem\"\n    }\n  ]\n}",
  "params": {
    "do_sample": {
      "type": "bool",
      "value": "false"
    },
    "max_tokens_to_generate": {
      "type": "int",
      "value": "100"
    },
    "repetition_penalty": {
      "type": "float",
      "value": "1"
    },
    "return_logits": {
      "type": "int",
      "value": "0"
    },
    "stop_sequences": {
      "type": "str",
      "value": ""
    },
    "temperature": {
      "type": "float",
      "value": "1"
    },
    "top_k": {
      "type": "int",
      "value": "50"
    },
    "top_p": {
      "type": "float",
      "value": "1"
    }
  }
}' 'http://localhost:3000/api/predict/generic/76ee2921-c54e-4988-9464-064a4face27e/200426b8-fb57-4535-9110-f68e923dba1b'

```

<img width="787" alt="image" src="https://github.com/user-attachments/assets/a059985e-98f2-4105-934b-a954b5781f21">

**Comment by AmitFeldman** [comment]
+1

**Comment by jf990** [comment]
I don't see the single quote as an issue, the single quoted strings are handled as expected. But is appears the parse is not properly handling the escaped single quote

you have this sequence:

`--data '{ "instan ... ent\": \"to a friend about a problem'\''? ANSWER: ...`

the `--data` string is opened with a single quote, so the first closing quote encountered closes that string. then there is an escaped single quote which isn't part of the curl command parameters so that might throw off the parser. then there is a new open quote for the next string.

if you change it to this sequence it works fine:

`--data '{ "instan ... ent\": \"to a friend about a problem? ANSWER: ...`

I'll look into the escaped single quote to see if we can fix that.

**Comment by jf990** [comment]
@avin45h @AmitFeldman this issue should be addressed. I added a few test cases for the escaped single quote and it seems to parse correctly now.

**Comment by AmitFeldman** [comment]
I don't think you published 1.4.0
